models/cycle_gan_semantic_model.py

```diff
@@ -150,13 +155,16 @@ def forward(self):
            #print('real_A shape=',self.real_A.shape)
            #print('real_A=',self.real_A)
            self.pred_real_A = self.netCLS(self.real_A)
-           _,self.gt_pred_A = self.pred_real_A.max(1)
-           pred_real_B = self.netCLS(self.real_B)
-           _,self.gt_pred_B = pred_real_B.max(1)
+           if not self.opt.regression:
+               _,self.gt_pred_A = self.pred_real_A.max(1)
```

Comment:
We should probably have a tensor here, what's the reason for taking the max ?

Reply:
it allows us to visualize the class predicted by the classifier (unused with a regressor)